std/traits.d

```diff
@@ -2902,42 +2874,55 @@ template RepresentationTypeTuple(T)
     static assert(R2.length == 2 && is(R2[0] == int) && is(R2[1] == immutable(Object)));
 }
 
-/*
-Statically evaluates to `true` if and only if `T`'s
-representation contains at least one field of pointer or array type.
-Members of class types are not considered raw pointers. Pointers to
-immutable objects are not considered raw aliasing.
-*/
-private template hasRawAliasing(T...)
+@safe unittest
 {
-    template Impl(T...)
+    struct VeryLargeType
     {
-        static if (T.length == 0)
+        import std.format : format;
+        import std.range : iota;
+
+        static foreach (i; 500.iota)
         {
-            enum Impl = false;
+            mixin(format!"int v%s;"(i));
```

Comment:
How does this affect the compilation time of the unittests?

Reply:
Compilation time of just `std.traits`, averaged over 10 runs:

branch | time | change
---------|------|-----
master | 2.475s +- 28ms | 0%
PR with unittest | 2.589s +- 50ms | 5% slower
PR without unittest | 2.411s +- 22ms | 7% faster

Reply:
So it has a well-noticable effect.
Can you reproduce the error also with lower n?

Reply:
No, because it only happens when we hit DMD's template recursion limit, which is 500: see src/dmd/dsymbolsem.d:5817.

Reply:
ping